src/components/AudioLevelIndicator/AudioLevelIndicator.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef } from 'react';
+import { AudioTrack, LocalAudioTrack, RemoteAudioTrack } from 'twilio-video';
+import MicOff from '@material-ui/icons/MicOff';
+import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
+
+let clipId = 0;
+const getUniqueClipId = () => clipId++;
+
+// @ts-ignore
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+let audioContext: AudioContext;
+
+function AudioLevelIndicator({
+  size,
+  audioTrack,
+  background,
+}: {
+  size?: number;
+  audioTrack?: AudioTrack;
+  background?: string;
+}) {
+  const SIZE = size || 24;
+  const ref = useRef<SVGRectElement>(null);
+  const isTrackEnabled = useIsTrackEnabled(audioTrack as LocalAudioTrack | RemoteAudioTrack);
+
+  useEffect(() => {
+    const SVGClipElement = ref.current;
+    let animationFrameID: number;
+
+    if (audioTrack && isTrackEnabled && SVGClipElement) {
+      audioContext = audioContext || new AudioContext();
+      const stream = new MediaStream();
+      stream.addTrack(audioTrack.mediaStreamTrack.clone());
+      const audioSource = audioContext.createMediaStreamSource(stream);
+
+      const analyser = audioContext.createAnalyser();
+      analyser.smoothingTimeConstant = 0.5;
+      analyser.fftSize = 1024;
+
+      audioSource.connect(analyser);
+
+      function render() {
+        animationFrameID = window.requestAnimationFrame(() => {
```

Comment:
requestAnimationFrame doesn't work if there are no repaints on the browser. For example, out of focus tabs will not get this. We solved this in voice by using a setTimeout

Reply:
We should also throttle this because this eats up a lot of CPU which makes the app slow. Very noticeable on low resource machines or phones.

Reply:
@charliesantos Actually, out of focus tabs need not repaint the audio level, so this is fine. Generally `setTimeout()` is not recommended for UI updates. Even `setTimeout` is throttled when a tab is backgrounded.

Reply:
Ok. This is probably fine since this is for UI. In our case as a SDK, customers wants to detect volume levels even when on background - out of focus, or when running as extension.

Reply:
I'm ok with this. We're using `requestAnimationFrame` for animating, so if the animations stop when the tab is out of focus that is ok. 

Also I'm not worried about performance. This kind of animation appears to be very efficient. I created a demo [here](https://winter-luck-gargoyle.glitch.me/) that creates 300 similar animations and it runs smoothly on my Galaxy 8. 

At most we would have 50 participants, and this would only animate those that have unmuted audio, which would be a number much lower than 50 in most cases.